src/Core/Client/Adapter/Http.php

```diff
@@ -71,15 +72,26 @@ public function check($data, $headers): void
     public function createContext($request, $endpoint)
     {
         $method = $request->getMethod();
-        $context = stream_context_create(
-            ['http' => [
-                    'method' => $method,
-                    'timeout' => $this->timeout,
-                    'protocol_version' => 1.0,
-                    'user_agent' => 'Solarium Http Adapter',
-                ],
-            ]
-        );
+        $httpOptions = [
+            'method' => $method,
+            'timeout' => $this->timeout,
+            'protocol_version' => 1.0,
+            'user_agent' => 'Solarium Http Adapter',
+        ];
+        if (null !== ($proxy = $this->getOption('proxy'))) {
+            $url = parse_url($proxy);
+            $url['scheme'] = 'tcp';
+            // fix behavior when only an IPv4 is passed as proxy, it would be parsed to path not to host
+            if (!isset($url['host']) && isset($url['path'])) {
+                $url['host'] = $url['path'];
+            }
+            if (!isset($url['port'])) {
+                $url['port'] = 8080;
+            }
+            $httpOptions['proxy']           =  sprintf('%s://%s:%d', $url['scheme'], $url['host'], $url['port']);
+            $httpOptions['request_fulluri'] = true;
```

Comment:
there's a lot of magic going on here which i think should be taken care of outside this library.
```php
if (null !== $proxy = $this->getProxy()) {
    $httpOptions['proxy'] = $proxy;
    $httpOptions['request_fulluri'] = true;
}
```
should be sufficient

Reply:
to clarify a bit more: this might work for your specific use case, but for instance makes it impossible to use a http proxy and probably will send those using a transparent proxy on a extended bughunt as the port is hardcoded to 8080 if not set.

the adapter should be responsible for configuring the stream option, not for any business logic surrounding it

Reply:
Okay, I aggree.
But what would be the best place to do this Options fixing logic ?
Do you thing it should be part of the configurable ?

Reply:
the best place would be when instantiating your adapter. so do your sprintf and port logic outside this library and instantiate your adapter with compiled options

Reply:
The reason to put this code there was, that the usage of this adapter is similar to the other adapters.
I think it would not be good, that each adapter needs different usage.

Reply:
?! the only adapter accepting a proxy option inside this library is the curl adapter which takes a single argument and does not apply random magic with protocols and ports. once more, i'm fine with a get / set proxy for the http adapter, but the added "logic" has to go...

Reply:
> 
> ?! the only adapter accepting a proxy option inside this library is the curl adapter which takes a single argument and does not apply random magic with protocols and ports. once more, i'm fine with a get / set proxy for the http adapter, but the added "logic" has to go...

Yes, I am talking about this.
I took the Code from curl adapter as example for this implementation here, that is the reason why I used the `ConfigurableInterface`.
And the stuff with the ports in no "magic", it is common knowledge, that port 8080 is the default port for proxy, if you don pass any custom ports.

Reply:
yess, but whenever someone decides to use 8080 for something else, let's say varnish cahe, this would be completely useless. there's no need for this library to decide which proxy port should be used

Reply:
This lib does not decide wich port is the best, this code yust makes in possible to pass ONLY hostname without port, if this is done, the defaultport is used.
And if you try to connect a proxy on a different port it is possible to add the port.

This is the expected behavior for a proxy setting

Reply:
> And the stuff with the ports in no "magic", it is common knowledge, that port 8080 is the default port for proxy, if you don pass any custom ports.

not sure I would say its "common knowledge" that port 8080 is a default proxy port. What's your source for that?

AFAIK its just an "alternative port" for http.

See 

https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?&page=109

Reply:
The most important refferences should be

https://de.wikipedia.org/wiki/Liste_der_standardisierten_Ports
and 
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?&page=120

There it is listed as http alternative, with the hint that this is mostly used for proxy.

And when you search in google https://www.google.de/search?q=http+proxy+default+port
you get many other sources telling the same.

Reply:
by your own logic; this library does not decide which is the best and does not set a default for something managed outside this library. so when you don't set a port it will be my.proxy.com and not my.proxy.com:8080

Reply:
That is the point. When you enter somewhere "this-address.com" in a http browser, it goes to default port 80,
When you enter it to an ftp software, you get default port 21, when you enter it in vnc you get default port 5900.

So in every software, a defaultport is choosen automatically, when you dont specify a port.
So why it should not be done in this lib ?